src/Psalm/Internal/Fork/PsalmRestarter.php

```diff
@@ -53,28 +63,54 @@ protected function requiresRestart($default): bool
             static fn(string $extension): bool => extension_loaded($extension)
         );
 
-        if (PHP_VERSION_ID >= 8_00_00 && (extension_loaded('opcache') || extension_loaded('Zend OPcache'))) {
+        $opcache_loaded = extension_loaded('opcache') || extension_loaded('Zend OPcache');
+
+        if (PHP_VERSION_ID >= 8_00_00 && $opcache_loaded) {
             // restart to enable JIT if it's not configured in the optimal way
-            if (!in_array(ini_get('opcache.enable_cli'), ['1', 'true', true, 1])) {
-                return true;
-            }
+            $opcache_settings = [
+                'enable_cli' => in_array(ini_get('opcache.enable_cli'), ['1', 'true', true, 1]),
+                'jit' => (int) ini_get('opcache.jit'),
+                'optimization_level' => (string) ini_get('opcache.optimization_level'),
+                'preload' => (string) ini_get('opcache.preload'),
+                'jit_buffer_size' => self::toBytes(ini_get('opcache.jit_buffer_size')),
+            ];
 
-            if (((int) ini_get('opcache.jit')) !== 1205) {
-                return true;
+            foreach (self::REQUIRED_OPCACHE_SETTINGS as $ini_name => $required_value) {
```

Comment:
Couldn't this be just an `if (self::REQUIRED_OPCACHE_SETTINGS !== $opcache_settings) { return true; }`?

Reply:
Perhaps, but that would require the keys in both arrays to be in the same order, which could easily be overlooked when adding/removing items.

Reply:
Makes sense!